python-backend/app/api/parties/party/resources/party.py

```diff
@@ -141,28 +141,34 @@ def get(self, mgr_appointment_guid):
     def post(self, mgr_appointment_guid=None):
         if mgr_appointment_guid:
             self.raise_error(400, 'Error: Unexpected manager id in Url.')
+
         data = ManagerResource.parser.parse_args()
         if not data['party_guid']:
             self.raise_error(400, 'Error: Party guid is not provided.')
         if not data['mine_guid']:
             self.raise_error(400, 'Error: Mine guid is not provided.')
         if not data['effective_date']:
             self.raise_error(400, 'Error: Effective date is not provided.')
-        # Validation for mine and party exists
-        party_exists = Party.find_by_party_guid(data['party_guid'])
-        if not party_exists:
+
+        new_party = Party.find_by_party_guid(data['party_guid'])
+        mine = MineIdentity.find_by_mine_guid(data['mine_guid'])
+        if not new_party:
             self.raise_error(400, 'Error: Party with guid: {}, does not exist.'.format(data['party_guid']))
-        mine_exists = MineIdentity.find_by_mine_guid(data['mine_guid'])
-        if not mine_exists:
+        if not mine:
             self.raise_error(400, 'Error: Mine with guid: {}, does not exist.'.format(data['mine_guid']))
-        # Check if there is a previous manager, set expiry the day before new manager
-        previous_mgr_expiry_date = data['effective_date'] - timedelta(days=1)
-        previous_mgrs = mine_exists.mgr_appointment
-        if previous_mgrs:
-            previous_mgr_info = previous_mgrs[0]
-            previous_mgr = MgrAppointment.find_by_mgr_appointment_guid(previous_mgr_info.mgr_appointment_guid)
+
+        # If there is a previous manager, set its expiry to the day before the new manager's start date
+        if mine.mgr_appointment:
+            previous_appointment_guid = mine.mgr_appointment[0].mgr_appointment_guid
+            previous_mgr = MgrAppointment.find_by_mgr_appointment_guid(previous_appointment_guid)
+
+            previous_mgr_expiry_date = (data['effective_date'] - timedelta(days=1)).date()
+            if previous_mgr_expiry_date < previous_mgr.effective_date:
+               self.raise_error(400, "Error: New manager's start date is on or before the previous manager's start date.")
```

Comment:
So, suppose that I add a manager to a mine, and set their start date to today. But they actually started yesterday.

If I try to correct that manager's start date, I'll get this error message. Is that what we want?

Reply:
But an implication of this fix for the history is that the current manager's start date can never be changed to an earlier date. So, if it is entered incorrectly once, it's permanently wrong.

This, to me, seems like trading one bug based on incorrect data input for another bug based on incorrect data input.

Reply:
Nathan, you're totally right. Those issues are tracked seperately by MDS-667 which would include more complex workflow changes. This item is a quick bug fix for the current workflow, but yes the current workflow has a poor user experience in that it doesn't handle erroneous input or adding historical data.